### PR TITLE
parser: trying to fix maxvalue list

### PR DESCRIPTION
**Pull request by zhexuany**
This PR aims to support query like the following:
```
CREATE TABLE rcx ( 
a INT, 
b INT, 
c CHAR(3), 
d INT ) 
PARTITION BY RANGE COLUMNS(a,d,c) ( 
PARTITION p0 VALUES LESS THAN (5,10,'ggg'), 
PARTITION p1 VALUES LESS THAN (10,20,'mmm'),
PARTITION p2 VALUES LESS THAN (15,30,'sss'), 
 PARTITION p3 VALUES LESS THAN (50,MAXVALUE,MAXVALUE)
);
```
and 
```
CREATE TABLE rcx ( 
a INT, 
b INT, 
c CHAR(3), 
d INT ) 
PARTITION BY RANGE COLUMNS(a,d,c) ( 
PARTITION p0 VALUES LESS THAN (5,10,'ggg'), 
PARTITION p1 VALUES LESS THAN (10,20,'mmm'),
PARTITION p2 VALUES LESS THAN (15,30,'sss'), 
 PARTITION p3 VALUES LESS THAN (MAXVALUE,MAXVALUE,MAXVALUE)
);
```

### CODE REVIEW

File: ast/expressions.go [hunk 1]
```diff
@@ -971,3 +971,22 @@ func (n *VariableExpr) Accept(v Visitor) (Node, bool) {
 	n.Value = node.(ExprNode)
 	return v.Leave(n)
 }
+
+// MaxValueExpr is the expression for "maxvalue" used in partition.
+type MaxValueExpr struct {
+	exprNode
+	Name string
```
Comment (coocood) [inline ast/expressions.go]: We don't need a name.

File: parser/parser.y [hunk 2]
```diff
@@ -1800,9 +1800,9 @@ PartitionDefinition:
 		}
 		switch $3.(type) {
 		case []ast.ExprNode:
-			partDef.LessThan = $3.([]ast.ExprNode)
-		case bool:
-			partDef.MaxValue = true
+						partDef.LessThan = $3.([]ast.ExprNode)
```
Comment (jackysp) [inline parser/parser.y]: Reset the indent.

File: parser/parser.y [hunk 3]
```diff
@@ -2183,6 +2179,10 @@ Expression:
 		$$ = &ast.IsNullExpr{Expr: $1, Not: !$2.(bool)}
 	}
 |	BoolPri
+|	"MAXVALUE"
```
[the remark this answers is not in the document]
Comment (zhexuany) [reply parser/parser.y]: gotcha.

File: ast/expressions.go (hunk 1 above)
Comment (zhexuany) [reply ast/expressions.go]: DONE

File: parser/parser.y (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhexuany) [reply parser/parser.y]: DONE

File: parser/parser.y (hunk 2 above)
Comment (zhexuany) [reply parser/parser.y]: DONE

**Comment by zhexuany** [comment]
@coocood @jackysp All comments are addressed. PTAL

File: parser/parser.y [hunk 4]
```diff
@@ -1800,9 +1802,9 @@ PartitionDefinition:
 		}
 		switch $3.(type) {
 		case []ast.ExprNode:
-			partDef.LessThan = $3.([]ast.ExprNode)
-		case bool:
-			partDef.MaxValue = true
+				partDef.LessThan = $3.([]ast.ExprNode)
+		case ast.ExprNode:
+				partDef.LessThan[0] = $3.(ast.ExprNode)
```
Comment (jackysp) [inline parser/parser.y]: I think the indents are still not right.

**Comment by coocood** [comment]
LGTM

**Pull request merged**